tests/5.0/requires/test_requires_unified_shared_memory.c

```diff
@@ -0,0 +1,136 @@
+//===---test_requires_unified_shared_memory.c -------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+//
+////===----------------------------------------------------------------------===//
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+#pragma omp requires unified_shared_memory
+
+int unified_shared_memory_scalar() {
+  OMPVV_INFOMSG("Unified shared memory testing - scalar value");
+  int errors = 0;
+  
+  int aVariable = 0;
+  // aVariable is only mapped to, with unified shared memory we should see the change reflected in the host.
+  // If no map is explicitely added, then scalars are firstprivate
+#pragma omp target map(to:aVariable)
+  {
+    aVariable += 10;
+  }
+
+  aVariable += 10;
+#pragma omp target map(alloc:aVariable)
```

Comment:
I'm not sure about the alloc here. What do you think should happen? (map is not necessarily a noop for scalars)

Reply:
It serves two purposes: 
1. It avoids the default behavior of `firstprivate` scalar variables, otherwise, even with shared memory it will be privatized
2. If an implementation is mistakenly not using shared memory, I expect this to cause an allocation that changes the  memory location of aVariable

Reply:
This code is invalid anyway.  All `map` clauses are required to have correct semantics even in a `unified_shared_memory` program, as the system is allowed to use them for optimization, so it's an invalid test.

Reply:
Actually both of these target regions are invalid tests.  The main things you get from this you can actually test are:

* can I access a host heap variable from the device
* can I access a host stack variable from the device
* can I access a host global variable from the device and is it *the same* variable as seen on the device
* can I access a target allocated variable from the host (think omp_target_alloc)

It also includes the requirements of `unified_address` so using `is_device_ptr` should work on any host pointer, and a zero-length map of an unmapped list-item should result in the original address being passed to the device.

Reply:
@trws Thanks for your input, really valuable. 

From what you are saying, if we use a pointer and clause like this:

```
int *aPtr = (int*) malloc(sizeof(int)*N);
#pragma omp target 
   for (int i =0; i < N; i ++) {
      aPtr[i] = i;
   }
```
It should work because the default map is zero length array?